app/_layout.tsx

```diff
@@ -26,42 +28,51 @@ function RootStack() {
     getPools();
   });
 
+  if (!privyAppId || !privyClientId) {
+    throw new Error('Privy App ID and Client ID are required');
+  }
+
   return (
-    <Stack screenOptions={{ headerShown: false }} initialRouteName="index">
-      <Stack.Screen name="index" options={{ title: 'AuthGate' }} />
+    <PrivyProvider appId={privyAppId} clientId={privyClientId}>
```

Comment:
More of a general question: why can't `PrivyProvider` be added to `AllProviders.tsx`?

Reply:
yeah done